File: backend/src/main/java/com/jelaniak/twittercloneproject/model/Tweet.java
package com.jelaniak.twittercloneproject.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.ser.std.ToStringSerializer;
import lombok.Data;
import org.bson.types.ObjectId;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Size;
import java.time.LocalDateTime;
import java.util.Set;

@Data
@Document(value = "Tweet")
public class Tweet {

    @Id
    @JsonSerialize(using = ToStringSerializer.class)
    private ObjectId tweetId;
    @JsonSerialize(using = ToStringSerializer.class)
    private ObjectId userId;
    private Media media;

    @NotBlank
    @Size(max = 280)
    private String content;

    @CreatedDate
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime dateOfCreation;

    @DBRef
    private Set<Comment> comments;

    private Integer commentCount;
    private Integer retweetCount;
    private Integer likeCount;

    private TweetType tweetType;
}

File: frontend/twitter-clone-ui/src/app/shared/models/tweet/tweet.ts
import { Media } from '../media/media';
import { Comment } from '../comment/comment';

export class Tweet {
  tweetId!: string;
  tweetUrl!: string;
  username!: string;
  media?: Media;
  content!: string;
  createdDate!: string;
  comments!: Set<Comment>;
  commentCount!: number;
  retweetCount!: number;
  likeCount!: number;
  tweetType!: TweetType;
}

enum TweetType {
  TWEET,
  RETWEET,
  MEDIA,
  LIKED,
}
